mountpoint-s3/src/prefetch/part_queue.rs

```diff
@@ -47,19 +55,22 @@ impl<E: std::error::Error + Send + Sync> PartQueue<E> {
         } else {
             // Do `try_recv` first so we can track whether the read is starved or not
             if let Ok(part) = self.receiver.try_recv() {
-                part
+                part.map_err(|e| e.into())
             } else {
                 let start = Instant::now();
-                let part = self.receiver.recv().await.expect("channel should not close");
+                let part = self.receiver.recv().await;
                 metrics::histogram!("prefetch.part_queue_starved_us", start.elapsed().as_micros() as f64);
-                part
+                match part {
+                    Err(_) => Err(PrefetchReadError::GetRequestTerminatedUnexpectedly),
```

Comment:
let's write the error explicitly on the LHS, in the very unlikely event `async_channel` ever gets more error cases